sdcm/utils/benchmarks.py

```diff
@@ -0,0 +1,252 @@
+    events_per_second: float = field(init=False)
+    latency_min: float = field(init=False)
+    latency_max: float = field(init=False)
+    latency_avg: float = field(init=False)
+    latency_p95: float = field(init=False)
+    thread_fairness_avg: float = field(init=False)
+    thread_fairness_stdev: float = field(init=False)
+    thread_fairness_time_avg: float = field(init=False)
+    thread_fairness_time_stdev: float = field(init=False)
+
+    def __post_init__(self):
+        eps_regex = re.compile(
+            r"events per second:\s+(?P<eps>[\d.]+)([\s\w\d.:/()]+)"
+            r"min:\s+(?P<latency_min>[\d.]+)([\s\w\d.:/()]+)"
+            r"avg:\s+(?P<latency_avg>[\d.]+)([\s\w\d.:/()]+)"
+            r"max:\s+(?P<latency_max>[\d.]+)([\s\w\d.:/()]+)"
+            r"95th percentile:\s+(?P<latency_p95>[\d.]+)([\s\w\d.:/()]+)"
+            r"events\s+\(avg/stddev\):\s+(?P<thread_avg>[\d.]+)/(?P<thread_stdev>[\d.]+)([\s\w\d.:/()]+)"
+            r"execution time \(avg/stddev\):\s+(?P<time_avg>[\d.]+)/(?P<time_stdev>[\d.]+)"
+        )
+        search_result = eps_regex.search(self.cmd_output)
+        self.events_per_second = search_result.groupdict()["eps"]
+        self.latency_min = search_result.groupdict()["latency_min"]
+        self.latency_max = search_result.groupdict()["latency_max"]
+        self.latency_avg = search_result.groupdict()["latency_avg"]
+        self.latency_p95 = search_result.groupdict()["latency_p95"]
+        self.thread_fairness_avg = search_result.groupdict()["thread_avg"]
+        self.thread_fairness_stdev = search_result.groupdict()["thread_stdev"]
+        self.thread_fairness_time_avg = search_result.groupdict()["time_avg"]
+        self.thread_fairness_time_stdev = search_result.groupdict()["time_stdev"]
+
+
+class ScyllaClusterBenchmarkManager(metaclass=Singleton):
+    """
+    ScyllaClusterBenchmarkManager gathers the benchmark results
+    of all the relevant db nodes in the cluster and presents
+    them in unified fashion.
+    ElasticSearch is used to store the results.
+    """
+
+    def __init__(self):
+        self._nodes: list["BaseNode"] = []
+        self._benchmark_runners: list[ScyllaNodeBenchmarkRunner] = []
+        self._es = ES()
+
+    def add_node(self, new_node: "BaseNode"):
+        if new_node.distro.is_debian_like:
+            self._benchmark_runners.append(ScyllaNodeBenchmarkRunner(new_node))
+        else:
+            LOGGER.info("Skipped installing benchmarking tools on a non-debian-like distro.")
+
+    def add_nodes(self, nodes: list["BaseNode"]):
+        for node in nodes:
+            self.add_node(node)
+
+    def install_benchmark_tools(self):
+        parallel = ParallelObject(self._benchmark_runners, timeout=30000)
+        parallel.run(lambda x: x.install_benchmark_tools())
+
+    def run_benchmarks(self):
+        parallel = ParallelObject(self._benchmark_runners, timeout=1200)
+        parallel.run(lambda x: x.run_benchmarks())
```

Comment:
This block of code gets changed in both other commits of this PR. Can you please combine these changes in one commit?

Reply:
Fixed (moved the change to the first commit)